src/internal/core.rs

```diff
@@ -18,20 +18,14 @@ use crate::version::Version;
 pub struct State<P: Package, V: Version> {
     root_package: P,
     root_version: V,
-
-    /// TODO: remove pub.
-    pub incompatibilities: Rc<Vec<Incompatibility<P, V>>>,
-
+    incompatibilities: Rc<Vec<Incompatibility<P, V>>>,
     /// Partial solution.
-    /// TODO: remove pub.
-    pub partial_solution: PartialSolution<P, V>,
-
+    pub(crate) partial_solution: PartialSolution<P, V>,
```

Comment:
I'm not a big fan of the `pub(crate)` thing as you may have noticed. I generally think that if I need something to be pub, everybody needs it. And I think the round of `pub(crate)` we did on the `term` module because I didn't spend enought time on the reporting API will byte us for #40 .

What I meant here by "remove pub" was "the outside world (outside this module)" should not know how this is done internally. And in fact #21 solves already two of those pubs by re-organizing indices. So we should maybe leave those out of this PR

Reply:
In addition, the `pub(crate)` here is equivalent to the `pub` anyway since `internal::` is not `pub`.

Reply:
Will revert.